File: app/src/main/java/com/example/well/clashroyaleguide/MainActivity.java
package com.example.well.clashroyaleguide;

import android.os.Bundle;
import android.support.design.widget.TabLayout;
import android.support.v4.app.FragmentManager;
import android.support.v4.view.ViewPager;
import android.support.v7.app.AppCompatActivity;

import com.example.well.clashroyaleguide.view.adapter.PageAdapter;

import butterknife.BindView;
import butterknife.ButterKnife;

public class MainActivity extends AppCompatActivity {

    @BindView(R.id.sliding_tabs)
    TabLayout tabLayout;
    @BindView(R.id.viewpager)
    ViewPager viewPager;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        ButterKnife.bind(this);

        setTabs();
    }

    private void setTabs() {
        FragmentManager fragmentManager = getSupportFragmentManager();

        tabLayout.removeAllTabs();

        tabLayout.addTab(tabLayout.newTab().setText("Arena"));
        tabLayout.addTab(tabLayout.newTab().setText("Cards"));
        tabLayout.addTab(tabLayout.newTab().setText("Decks"));
        tabLayout.addTab(tabLayout.newTab().setText("Player"));
        tabLayout.setTabGravity(TabLayout.GRAVITY_FILL);

        final PageAdapter pageAdapter = new PageAdapter(fragmentManager);
        viewPager.setAdapter(pageAdapter);
        viewPager.setOffscreenPageLimit(tabLayout.getTabCount());
        viewPager.setCurrentItem(0);
        viewPager.addOnPageChangeListener(new TabLayout.TabLayoutOnPageChangeListener(tabLayout));
        tabLayout.addOnTabSelectedListener(new TabLayout.OnTabSelectedListener() {
            @Override
            public void onTabSelected(TabLayout.Tab tab) {
                viewPager.setCurrentItem(tab.getPosition());
            }

            @Override
            public void onTabUnselected(TabLayout.Tab tab) {

            }

            @Override
            public void onTabReselected(TabLayout.Tab tab) {

            }
        });
    }

}

File: app/src/main/java/com/example/well/clashroyaleguide/MainActivity.kt
package com.example.well.clashroyaleguide

import android.os.Bundle
import android.support.design.widget.BottomNavigationView
import android.support.v4.app.Fragment
import android.support.v7.app.ActionBar
import android.support.v7.app.AppCompatActivity
import com.example.well.clashroyaleguide.card.CardFragment
import com.example.well.clashroyaleguide.deck.DeckFragment
import com.example.well.clashroyaleguide.player.PlayerFragment
import kotlinx.android.synthetic.main.activity_main.*

/**
 * Created by wellingtonyogui on 18/09/2018 for guideRoayle-android.
 */
class MainActivity: AppCompatActivity() {
    lateinit var toolbar: ActionBar

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        supportActionBar?.let {
            toolbar = it
        }

        val bottomNavigation: BottomNavigationView = navigationView
        bottomNavigation.setOnNavigationItemSelectedListener(onNavigationItemSelectedListener)
    }

    private val onNavigationItemSelectedListener = BottomNavigationView.OnNavigationItemSelectedListener { item ->
        when (item.itemId) {
            //TODO: Implementar busca,
            R.id.navigation_player -> {
                toolbar.title = "Player"
                val playerFragment = PlayerFragment.newInstance()
                openFragment(playerFragment)
                return@OnNavigationItemSelectedListener true
            }
            R.id.navigation_cards -> {
                toolbar.title = "Cartas"
                val cardFragment = CardFragment.newInstance()
                openFragment(cardFragment)
                return@OnNavigationItemSelectedListener true
            }
            R.id.navigation_decks -> {
                toolbar.title = "Decks"
                val deckFragment = DeckFragment.newInstance()
                openFragment(deckFragment)
                return@OnNavigationItemSelectedListener true
            }
        }
        false
    }

    private fun openFragment(fragment: Fragment) {
        val transaction = supportFragmentManager.beginTransaction()
        transaction.replace(R.id.container, fragment)
        transaction.commit()
    }
}